torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -251,7 +252,10 @@ def _share_state_and_init_handle_attrs(
     root_state._all_fsdp_states = traversal_utils._get_fsdp_states(root_module)
     root_state._all_handles = root_state._exec_order_data.all_handles  # share reference
     root_state._device_mesh = _init_device_mesh(root_state)
-
+    import os
+    root_state._full_prec_in_eval = (
+        os.environ.get(_FSDP_FULL_PREC_IN_EVAL, "") == "1"
+    )
```

Comment:
We should move this to construction time. For example, we can check the new env var after this:
https://github.com/pytorch/pytorch/blob/adf1405909a3136b51be17992a8def4297764022/torch/distributed/fsdp/_init_utils.py#L393-L397